whosonfirst/whosonfirst-www-boundaryissues
# Offline saving (re: #113)

Ok, I think this is ready! Here's what it does, described in greater detail in [lib_wof_save.php](https://github.com/whosonfirst/whosonfirst-www-boundaryissues/blob/dphiffer/offline-saving/www/include/lib_wof_save.php).

Here's how to test it out:
1. Save some things in the web UI
2. Go to `/usr/local/mapzen/whosonfirst-www-boundaryissues/pending` and have a look around
3. Run `sudo -u www-data php /usr/local/mapzen/whosonfirst-www-boundaryissues/bin/save_pending.php` to add/commit/push to GitHub

That last step is running in the range of 7 or 8 minutes for me, so give it some time.

Some questions:
- I modified `wof_utils_id2abspath` to optionally take an array as the `$root` argument, in order to check multiple possible data trees, but I wonder if I should have created a new function entirely?
- Should I do a `git diff` after [pulling in changes](https://github.com/whosonfirst/whosonfirst-www-boundaryissues/blob/dphiffer/offline-saving/www/include/lib_wof_save.php#L349-L356) from GitHub and update Elasticsearch for the incoming changes?
- Currently edits by multiple people will be "mis-authored" by whoever happened to edit the last pending record, should I fix that?
